loader/loader_mock.go

```diff
@@ -262,3 +263,11 @@ func (l *mockLoader) GetPipelineRun(c client.Client, ctx context.Context, name,
 	}
 	return toolkit.GetMockedResourceAndErrorFromContext(ctx, GetPipelineRunContextKey, &tektonv1.PipelineRun{})
 }
+
+// GetComponent returns the resource and error passed as values of the context.
+func (l *mockLoader) GetComponent(c client.Client, ctx context.Context, name, namespace string) (*applicationapiv1alpha1.Component, error) {
```

Comment:
Note: this function will need unit test coverage in [loader_mock_test.go](https://github.com/redhat-appstudio/integration-service/blob/main/loader/loader_mock_test.go)